src/nbl/video/ILogicalDevice.cpp

```diff
@@ -469,8 +472,13 @@ bool ILogicalDevice::updateDescriptorSets(const std::span<const IGPUDescriptorSe
                 return false;
         }
         outCategory++;
+        i++;
     }
 
+    core::vector<redirect_t::storage_range_index_t> copySrcDescriptorRedirectIndices(descriptorCopies.size()), copySrcMutableSamplerRedirectIndices(descriptorCopies.size());
+    core::vector<redirect_t::storage_range_index_t> copyDstDescriptorRedirectIndices(descriptorCopies.size()), copyDstMutableSamplerRedirectIndices(descriptorCopies.size());
+    core::vector<asset::IDescriptor::E_TYPE> copyTypes(descriptorCopies.size());
```

Comment:
normally SoA perform better than AoS, except when the arrays are backed by separate dynamic allocations/containers

Hence moving to a Array of Structures and having the validation parameters in one struct will benefit you